### PR TITLE
Advancement DoABarrelRoll and Updated PlayerCraftItemEvent

### PR DESCRIPTION
**Pull request by JulixnDev**
Added: Advancement DoABarrelRoll \
Feat: Improved MixinResultSlot and PlayerCraftItemEvent

### CODE REVIEW

File: src/main/java/live/einfachgustaf/mods/smp/mixins/MixinResultSlot.java [hunk 1]
```diff
@@ -5,16 +5,33 @@
...
 public class MixinResultSlot {
 
+    @Shadow
+    @Final
+    private Player player;
+
+    private boolean wasQuickCrafted = false;
+
     @Inject(at = @At("HEAD"), method = "onTake")
     public void injectSlotsChanged(Player player, ItemStack itemStack, CallbackInfo ci) {
```
Comment (DinoMarlir) [inline src/main/java/live/einfachgustaf/mods/smp/mixins/MixinResultSlot.java:25]: please rename the method to injectOnTake

**Review by DinoMarlir — CHANGES_REQUESTED** [review]
only one suggestion, which is not essential but very well done! thank you. :)

File: src/main/java/live/einfachgustaf/mods/smp/mixins/MixinResultSlot.java (hunk 1 above)
Comment (JulixnDev) [reply src/main/java/live/einfachgustaf/mods/smp/mixins/MixinResultSlot.java:25]: This is how I got it from main

Comment (DinoMarlir) [reply src/main/java/live/einfachgustaf/mods/smp/mixins/MixinResultSlot.java:25]: oh, than it's my fault. sorry

**Review by DinoMarlir — COMMENTED** [review]
lgtm

**Review by DinoMarlir — COMMENTED** [review]
yo

**Pull request merged**